src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const stateHistory = [];
+
+  for (let index = 0; index < actions.length; index++) {
+    const action = actions[index];
+    const type = action.type;
+
+    switch (type) {
+      case 'addProperties':
+        stateCopy = { ...stateCopy, ...action.extraData };
+        stateHistory.push({ ...stateCopy });
```

Comment:
According to the checklist, it is better to push a copy of the object to the array at the end of the each loop cycle, but outside of the `switch` block, after the current action is processed. This avoids repetition and ensures that the state copy is always added to the history after any action.